src/backend/distributed/commands/role.c

```diff
@@ -275,3 +514,236 @@ makeDefElemInt(char *name, int value)
 {
 	return makeDefElem(name, (Node *) makeInteger(value), -1);
 }
+
+
+/*
+ * GetDatabaseNameFromDbRoleSetting performs a lookup, and finds the database name
+ * associated with a Role Setting
+ */
+static char *
+GetDatabaseNameFromDbRoleSetting(HeapTuple tuple, TupleDesc DbRoleSettingDescription)
+{
+	bool isnull;
+
+	Datum setdatabase = heap_getattr(tuple, Anum_pg_db_role_setting_setdatabase,
+									 DbRoleSettingDescription, &isnull);
+
+	if (isnull)
+	{
+		return NULL;
+	}
+
+	Oid databaseId = DatumGetObjectId(setdatabase);
+	char *databaseName = get_database_name(databaseId);
+
+	return databaseName;
+}
+
+
+/*
+ * GetDatabaseNameFromDbRoleSetting performs a lookup, and finds the role name
+ * associated with a Role Setting
+ */
+static char *
+GetRoleNameFromDbRoleSetting(HeapTuple tuple, TupleDesc DbRoleSettingDescription)
+{
+	bool isnull;
+
+	Datum setrole = heap_getattr(tuple, Anum_pg_db_role_setting_setrole,
+								 DbRoleSettingDescription, &isnull);
+
+	if (isnull)
+	{
+		return NULL;
+	}
+
+	Oid roleId = DatumGetObjectId(setrole);
+	char *roleName = GetUserNameFromId(roleId, true);
+
+	return roleName;
+}
+
+
+/*
+ * MakeSetStatementArgs parses a configuraton value and creates an A_Const
+ * with an appropriate type.
+ *
+ * The allowed A_Const types are Integer, Float, and String.
+ */
+Node *
+MakeSetStatementArgument(char *configurationName, char *configurationValue)
+{
+	Node *arg = NULL;
+	char **key = &configurationName;
+	struct config_generic **gucVariables = get_guc_variables();
```

Comment:
I'd love a small comment above this function that we are using a function which postgres' intend for the export was specifically NOT to be used, but that this is the only way for us to find the internal type of a guc which is important for recreation of the SET statement.

Reply:
```c

	/* Perform a lookup on GUC variables to find the config type and units.
	 * All user-defined GUCs have string values, but we need to perform a search
	 * on all the GUCs to understand if it is a user-defined one or not.
	 *
	 * Note: get_guc_variables() is intended for internal use only, but there
	 * is no other way to determine allowed units, and value types other than
	 * using this function
	 */
```